panel/applets/icon-tasklist/WindowManager.vala

```diff
@@ -0,0 +1,19 @@
+/*
+ * This file is part of budgie-desktop
+ *
+ * Copyright (C) 2016 Fernando Mussel <fernandomussel91@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+public interface WindowManager
```

Comment:
yeah, you have a point. I used an interface to avoid coupling the classes to a concrete implementation (IconTasklistApplet in this case). But at the moment, this is an unnecessary complexity.